experimental/SetPartitions/src/Util.jl

```diff
@@ -0,0 +1,160 @@
+"""
+    _new_point_values(p_upper::Vector{Int}, p_lower::Vector{Int}, 
+                      q_upper::Vector{Int}, q_lower::Vector{Int})
+
+Return two vectors which are semantically identical to `q_upper` and `q_lower`
+as set-partition and have no common values with `p_upper` and `p_lower`.
+"""
+function _new_point_values(p_upper::Vector{Int}, p_lower::Vector{Int}, 
+                          q_upper::Vector{Int}, q_lower::Vector{Int})
+
+    p_points = vcat(p_upper, p_lower)
+
+    if !(isempty(p_points))
+        new_id, index = findmax(p_points)
+        new_id += 1
+    else
+        new_id = 1 
+    end
+    
+    q_points = vcat(q_upper, q_lower)
+    new_ids = Dict{Int, Int}()
+
+    for n in q_points
+        new_ids[n] = new_id
+        new_id += 1
+    end
+    
+    upper = [new_ids[n] for n in q_upper]
+    lower = [new_ids[n] for n in q_lower]
+    
+    return (upper, lower)
+end
+
+"""
+    _normal_form(p_upper::Vector{Int}, p_lower::Vector{Int})
+
+Return two vectors which are semantically identical to `p_upper` and `p_lower`
+as set-partition and are numbered from 1 to the number of subsets.
+"""
+function _normal_form(p_upper::Vector{Int}, p_lower::Vector{Int})
+
+    new_id = 1
+    new_ids = Dict{Int, Int}()
+    p_return = _new_point_values([length(p_upper) + length(p_lower)], 
+        Vector{Int}(), p_upper, p_lower)
+
+    for (i, n) in enumerate(p_return[1])
+        if !(n in keys(new_ids))
+            new_ids[n] = new_id
+            p_return[1][i] = new_id
+            new_id += 1
+        else
+            p_return[1][i] = new_ids[n]
+        end
+    end
+
+    for (i, n) in enumerate(p_return[2])
+        if !(n in keys(new_ids))
+            new_ids[n] = new_id
+            p_return[2][i] = new_id
+            new_id += 1
+        else
+            p_return[2][i] = new_ids[n]
+        end
+    end
+
+    return p_return
+end
+
+"""
+    _req_colored_partition(partition::AbstractPartition, 
+                            color_upper_points::Vector{Int}, 
+                            color_lower_points::Vector{Int})
+
+Return true if the format of the attributes for initializing a
+colored partition is valid, false otherwise.
+
+Note that this is a helper function for the constructor of colored partitions.
+"""
+function _req_colored_partition(partition::AbstractPartition, 
```

Comment:
I, personally, wouldn't have these as a separate function and just inline them into the constructor. This makes it furthermore possible to use multiple `@req` statements with different error messages tho help a user with wrong input.